### PR TITLE
Add ray tracing acceleration structure build experimental extension

### LINKED ISSUE #146: Add extension for ray tracing acceleration structure build
This extension should provide a sufficiently abstract API for building ray tracing acceleration structures, which in turn may be used on device for hardware accelerated ray tracing.

### PR DESCRIPTION
**Pull request by wdamon-intel**
This extension provides the functionality to build ray tracing acceleration structures for 3D scenes on the host for use with GPU devices.

Resolves #146

### CODE REVIEW

File: scripts/core/RTASBuilder.yml [hunk 1]
```diff
@@ -0,0 +1,959 @@
...
+      name: geometryMask
+      desc: "[in] 8-bit geometry mask for ray masking"
+    - type: uint8_t
+      name: reserved
+      desc: "[in] reserved for future use"
+    - type: uint32_t
+      name: primCount
+      desc: "[in] number of primitives in geometry"
+    - type: $x_rtas_geometry_aabbs_cb_exp_t
+      name: pfnGetBoundsCb
```
Comment (jandres742) [inline scripts/core/RTASBuilder.yml:680]: do we have in the programmers guide a sample on how the callback is used?

File: scripts/core/RTASBuilder.yml [hunk 2]
```diff
@@ -0,0 +1,959 @@
...
+      name: pRtasBuffer
+      desc: "[in] pointer to destination buffer"
+    - type: size_t
+      name: rtasBufferSizeBytes
+      desc: "[in] destination buffer size, in bytes"
+    - type: $x_rtas_parallel_operation_exp_handle_t
+      name: hParallelOperation
+      desc: "[in][optional] handle to parallel operation object"
+    - type: void*
+      name: pBuildUserPtr
```
Comment (jandres742) [inline scripts/core/RTASBuilder.yml:839]: same about the sample

File: scripts/core/common.yml [hunk 3]
```diff
@@ -544,6 +578,18 @@ members:
...
 type: struct
+desc: "Base for all callback function parameter types"
+name: $x_base_cb_params_t
+members:
+    - type: $x_structure_type_t
+      name: stype
+      desc: "[in] type of this structure"
+    - type: "void*"
+      name: pNext
+      desc: "[in,out][optional] must be null or a pointer to an extension-specific structure (i.e. contains sType and pNext)."
```
Comment (jandres742) [inline scripts/core/common.yml:589]: sType -> stype

File: third_party/requirements.txt [hunk 4]
```diff
@@ -11,7 +11,7 @@ exhale==0.3.0
 idna==2.8
 imagesize==1.1.0
 Jinja2==2.10.3
-lxml==4.9.1
+lxml==4.9.2
```
Comment (jandres742) [inline third_party/requirements.txt:14]: why this change?

Comment (wdamon-intel) [reply third_party/requirements.txt:14]: Windows seems happier with this version. That's the only reason, TBH.

File: scripts/core/RTASBuilder.yml (hunk 1 above)
Comment (wdamon-intel) [reply scripts/core/RTASBuilder.yml:680]: The programming guide does not yet include an example using this callback. Will look into adding one.

File: scripts/core/common.yml (hunk 3 above)
Comment (wdamon-intel) [reply scripts/core/common.yml]: Fixed

File: scripts/core/RTASBuilder.yml (hunk 1 above)
Comment (wdamon-intel) [reply scripts/core/RTASBuilder.yml:680]: @jandres742 After looking into it, my feeling is that using these features requires a more elaborate example than maybe the introductory programming guide can trivially provide. We will file a separate issue to expand the programming guide with an **Advanced Concepts** chapter in the next release.
